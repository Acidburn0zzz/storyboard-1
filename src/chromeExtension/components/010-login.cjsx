React             = require 'react'
ReactRedux        = require 'react-redux'
actions           = require '../actions/actions'

RETURN_KEY = 13

mapStateToProps = ({cx: {fLoginRequired, loginState, login}}) -> 
  return {fLoginRequired, loginState, login}
mapDispatchToProps = (dispatch) ->
  onLogIn: (credentials) -> dispatch actions.logIn credentials
  onLogOut: -> dispatch actions.logOut()

Login = React.createClass
  displayName: 'Login'

  #-----------------------------------------------------
  propTypes:
    # From Redux.connect
    fLoginRequired:         React.PropTypes.bool.isRequired
    loginState:             React.PropTypes.string.isRequired
    login:                  React.PropTypes.string
    onLogIn:                React.PropTypes.func.isRequired
    onLogOut:               React.PropTypes.func.isRequired
  getInitialState: ->
    login:                  ''
    password:               ''

  #-----------------------------------------------------
  render: -> 
    {fLoginRequired, loginState} = @props
    if not fLoginRequired
      return <div style={_style.outer()}><i>No login required to see server logs</i></div>
    if loginState is 'LOGGED_IN'
      return @renderLogOut()
    else
      return @renderLogIn()

  renderLogOut: ->
    <div style={_style.outer()}>
      Logged in as {@props.login}
      {' '}
      <button onClick={@logOut}>Log out</button>
    </div>

  renderLogIn: ->
    {loginState} = @props
    btnMessage = switch loginState
      when 'LOGGED_OUT' then 'Submit'
      when 'LOGGING_IN' then 'Logging in...'
      else 'Logged in'
    <div style={_style.outer true}>
      <b>Log in to see server logs:</b>
      {' '}
      <span>
        <input ref="login"
          id="login"
          type="text"
          value={@state.login}
          placeholder="Login"
          onChange={@onChangeCredentials}
          onKeyUp={@onKeyUpCredentials}
          style={_style.field}
        />
        <input ref="password"
          id="password"
          type="password"
          value={@state.password}
          placeholder="Password"
          onChange={@onChangeCredentials}
          onKeyUp={@onKeyUpCredentials}
          style={_style.field}
        />
        <button 
          onClick={@logIn} 
          disabled={loginState isnt 'LOGGED_OUT'}
        >
          {btnMessage}
        </button>
      </span>
    </div>

  #-----------------------------------------------------
  logIn: -> @props.onLogIn @state 
  logOut: ->
    @setState {login: '', password: ''}
    @props.onLogOut()

  onKeyUpCredentials: (ev) -> @logIn() if ev.which is RETURN_KEY

  onChangeCredentials: (ev) -> 
    @setState {"#{ev.target.id}": ev.target.value}

#-----------------------------------------------------
_style = 
  outer: (fHighlight) ->
    padding: "4px 4px 4px 10px"
    backgroundColor: if fHighlight then '#d6ecff'
  field:
    marginRight: 4
    width: 70

#-----------------------------------------------------
connect = ReactRedux.connect mapStateToProps, mapDispatchToProps
module.exports = connect Login
