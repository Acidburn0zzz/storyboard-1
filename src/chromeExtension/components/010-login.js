import React from 'react';
import * as ReactRedux from 'react-redux';
import {
  TextInput, PasswordInput,
  Icon, Spinner,
} from 'giu';
import Promise from 'bluebird';
import actions from '../actions/actions';

const RETURN_KEY = 13;

const mapStateToProps = ({ cx: { fLoginRequired, loginState, login } }) => ({
  fLoginRequired,
  loginState,
  login,
});

class Login extends React.PureComponent {
  static propTypes = {
    colors: React.PropTypes.object.isRequired,
    fLoginRequired: React.PropTypes.bool,
    loginState: React.PropTypes.string.isRequired,
    login: React.PropTypes.string,
    logIn: React.PropTypes.func.isRequired,
    logOut: React.PropTypes.func.isRequired,
  };

  // -----------------------------------------------------
  render() {
    const { fLoginRequired, loginState, colors } = this.props;
    if (fLoginRequired == null) {
      return (
        <div style={style.outer(colors)}>
          <Spinner size="lg" fixedWidth />
        </div>
      );
    }
    if (!fLoginRequired) {
      return <div style={style.outer(colors)}><i>No login required to see server logs</i></div>;
    }
    return loginState === 'LOGGED_IN'
      ? this.renderLogOut()
      : this.renderLogIn();
  }

  renderLogOut() {
    const { login, colors } = this.props;
    const msg = login ? `Logged in as ${login}` : 'Logged in';
    return (
      <div style={style.outer(colors)}>
        {msg}
        {' '}
        <Icon
          icon="sign-out"
          title="Log out"
          size="lg"
          fixedWidth
          onClick={this.logOut}
        />
      </div>
    );
  }

  renderLogIn() {
    const { loginState, colors } = this.props;
    let btn;
    switch (loginState) {
      case 'LOGGED_OUT':
        btn = (
          <Icon
            icon="sign-in"
            title="Log in"
            size="lg"
            fixedWidth
            onClick={this.logIn}
          />
        );
        break;
      case 'LOGGING_IN':
        btn = <Spinner title="Logging in" size="lg" fixedWidth />;
        break;
      default:
        btn = '';
        break;
    }
    return (
      <div style={style.outer(colors, true)}>
        <b>Server logs:</b>
        {' '}
        <TextInput
          ref="login"
          id="login"
          placeholder="Login"
          onKeyUp={this.onKeyUpCredentials}
          style={style.field}
          required errorZ={12}
        />
        <PasswordInput
          ref="password"
          id="password"
          placeholder="Password"
          onKeyUp={this.onKeyUpCredentials}
          style={style.field}
          required errorZ={12}
        />
        {btn}
      </div>
    );
  }

  // -----------------------------------------------------
  logIn = () => {
    const credentials = {};
    Promise.map(['login', 'password'], (key) =>
      this.refs[key].validateAndGetValue()
      .then((val) => { credentials[key] = val; })
    )
    .then(() => this.props.logIn(credentials));
  }

  logOut = () => { this.props.logOut(); }

  onKeyUpCredentials = (ev) => {
    if (ev.which !== RETURN_KEY) return;
    this.logIn();
  }
}

// -----------------------------------------------------
const style = {
  outer: (colors, fHighlight) => ({
    padding: '4px 10px',
    backgroundColor: fHighlight ? colors.colorServerBg : colors.colorUiBg,
    color: fHighlight ? colors.colorServerFg : colors.colorUiFg,
  }),
  field: {
    marginRight: 4,
    width: 70,
    backgroundColor: 'transparent',
  },
};

// -----------------------------------------------------
const connect = ReactRedux.connect(mapStateToProps, actions);
export default connect(Login);
export { Login as _Login };
