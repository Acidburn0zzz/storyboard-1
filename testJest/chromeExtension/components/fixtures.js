import { addDefaults } from 'timm';

export const BASE_COLORS = {
  colorClientBg: 'aliceblue',
  colorClientBgIsDark: false,
  colorClientFg: 'black',
  colorServerBg: 'rgb(214, 236, 255)',
  colorServerBgIsDark: false,
  colorServerFg: 'black',
  colorUiBg: 'white',
  colorUiBgIsDark: false,
  colorUiFg: 'rgb(64, 64, 64)',
};

export const DARK_COLORS = {
  colorClientBg: 'rgb(17, 22, 54)',
  colorClientBgIsDark: true,
  colorClientFg: 'white',
  colorServerBg: 'rgb(14, 11, 33)',
  colorServerBgIsDark: true,
  colorServerFg: 'white',
  colorUiBg: 'black',
  colorUiBgIsDark: true,
  colorUiFg: 'white',
};

export const EMPTY_MAIN_STORY = {
  fExpanded: true,
  fHierarchical: true,
  fOpen: true,
  fWrapper: true,
  records: [
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: false,
      fStoryObject: true,
      id: 'main_0',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/0',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_0',
      t: 0,
      title: 'Client',
    },
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: true,
      fStoryObject: true,
      id: 'main_1',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/1',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_1',
      t: 0,
      title: 'Server',
    },
  ],
};

export const buildLogRecord = (options = {}) => addDefaults(options, {
  id: 'id1',
  hubId: 'hubId1',
  version: 3,
  fStory: false,
  fServer: false,
  storyId: 'story1',
  t: 0,
  src: 'main',
  level: 30,
  msg: 'Example message',
});

export const buildStory = (options = {}) => {
  const story = addDefaults(options, {
    id: 'id1',
    hubId: 'hubId1',
    version: 3,
    fStory: true,
    fServer: false,
    storyId: 'story1',
    t: 0,
    src: 'main',
    level: 30,
    fRoot: false,
    title: 'Story title',
    parents: [],
    // client-side
    fStoryObject: true,
    fOpen: false,
    fMain: false,
    status: undefined,
    fExpanded: true,
    fHierarchical: true,
    records: [],
  });
  story.records.unshift({
    id: `${story.id}.creation`,
    hubId: story.hubId,
    version: story.version,
    fStory: true,
    fServer: story.fServer,
    storyId: story.storyId,
    t: story.t,
    src: story.src,
    level: story.level,
    fRoot: story.fRoot,
    title: story.title,
    action: 'CREATED',
  });
  if (!story.fOpen) {
    story.records.push({
      id: `${story.id}.closure`,
      hubId: story.hubId,
      version: story.version,
      fStory: true,
      fServer: story.fServer,
      storyId: story.storyId,
      t: story.records[story.records.length - 1].t,
      src: story.src,
      level: story.level,
      fRoot: story.fRoot,
      title: story.title,
      action: 'CLOSED',
    });
  }
  story.numRecords = story.records.length;
  return story;
};
