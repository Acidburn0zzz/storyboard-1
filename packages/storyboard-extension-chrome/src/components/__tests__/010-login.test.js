/* eslint-env jest */
/* eslint-disable no-extend-native, react/jsx-no-duplicate-props */
import React from 'react';
import renderer from 'react-test-renderer';
import { Floats } from 'giu';
import { _Login as Login } from '../010-login';
import { BASE_COLORS } from './fixtures';

jest.mock('react-dom');

describe('Login', () => {
  it('renders correctly when login needs are still unknown', () => {
    const tree = renderer.create(
      <Login
        colors={BASE_COLORS}
        loginState="LOGGED_OUT"
        logIn={() => {}}
        logOut={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when no login is required', () => {
    const tree = renderer.create(
      <Login
        colors={BASE_COLORS}
        fLoginRequired={false}
        loginState="LOGGED_OUT"
        logIn={() => {}}
        logOut={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when LOGGED_IN', () => {
    const tree = renderer.create(
      <Login
        colors={BASE_COLORS}
        fLoginRequired
        loginState="LOGGED_IN"
        login="Guille"
        logIn={() => {}}
        logOut={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when LOGGED_OUT', () => {
    const tree = renderer.create(
      <div>
        <Floats />
        <Login
          colors={BASE_COLORS}
          fLoginRequired
          loginState="LOGGED_OUT"
          logIn={() => {}}
          logOut={() => {}}
        />
      </div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when LOGGING_IN', () => {
    const tree = renderer.create(
      <div>
        <Floats />
        <Login
          colors={BASE_COLORS}
          fLoginRequired
          loginState="LOGGING_IN"
          logIn={() => {}}
          logOut={() => {}}
        />
      </div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
