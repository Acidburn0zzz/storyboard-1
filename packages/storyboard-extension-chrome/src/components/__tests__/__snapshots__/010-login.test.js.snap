exports[`Login renders correctly when LOGGED_IN 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "color": "rgb(64, 64, 64)",
      "padding": "4px 10px",
    }
  }>
  Logged in as Guille
   
  <i
    className="giu-icon fa fa-sign-out fa-lg fa-fw"
    onClick={[Function]}
    style={
      Object {
        "color": undefined,
        "cursor": "pointer",
        "letterSpacing": "normal",
      }
    }
    title="Log out" />
</div>
`;

exports[`Login renders correctly when LOGGED_OUT 1`] = `
<div>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    } />
  <div
    style={
      Object {
        "backgroundColor": "rgb(214, 236, 255)",
        "color": "black",
        "padding": "4px 10px",
      }
    }>
    <b>
      Server logs:
    </b>
     
    <input
      className="giu-text-input"
      id="login"
      onBlur={[Function]}
      onChange={[Function]}
      onCopy={undefined}
      onCut={undefined}
      onFocus={[Function]}
      onKeyUp={[Function]}
      onPaste={undefined}
      orient={undefined}
      placeholder="Login"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "1px solid #999999",
          "color": "inherit",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "marginRight": 4,
          "width": 70,
        }
      }
      tabIndex={undefined}
      type="text"
      value="" />
    <input
      className="giu-password-input"
      id="password"
      onBlur={[Function]}
      onChange={[Function]}
      onCopy={undefined}
      onCut={undefined}
      onFocus={[Function]}
      onKeyUp={[Function]}
      onPaste={undefined}
      orient={undefined}
      placeholder="Password"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "1px solid #999999",
          "color": "inherit",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "marginRight": 4,
          "width": 70,
        }
      }
      tabIndex={undefined}
      type="password"
      value="" />
    <i
      className="giu-icon fa fa-sign-in fa-lg fa-fw"
      onClick={[Function]}
      style={
        Object {
          "color": undefined,
          "cursor": "pointer",
          "letterSpacing": "normal",
        }
      }
      title="Log in" />
  </div>
</div>
`;

exports[`Login renders correctly when LOGGING_IN 1`] = `
<div>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    } />
  <div
    style={
      Object {
        "backgroundColor": "rgb(214, 236, 255)",
        "color": "black",
        "padding": "4px 10px",
      }
    }>
    <b>
      Server logs:
    </b>
     
    <input
      className="giu-text-input"
      id="login"
      onBlur={[Function]}
      onChange={[Function]}
      onCopy={undefined}
      onCut={undefined}
      onFocus={[Function]}
      onKeyUp={[Function]}
      onPaste={undefined}
      orient={undefined}
      placeholder="Login"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "1px solid #999999",
          "color": "inherit",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "marginRight": 4,
          "width": 70,
        }
      }
      tabIndex={undefined}
      type="text"
      value="" />
    <input
      className="giu-password-input"
      id="password"
      onBlur={[Function]}
      onChange={[Function]}
      onCopy={undefined}
      onCut={undefined}
      onFocus={[Function]}
      onKeyUp={[Function]}
      onPaste={undefined}
      orient={undefined}
      placeholder="Password"
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "1px solid #999999",
          "color": "inherit",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "marginRight": 4,
          "width": 70,
        }
      }
      tabIndex={undefined}
      type="password"
      value="" />
    <i
      className="giu-icon fa fa-circle-o-notch fa-lg fa-fw fa-spin"
      style={
        Object {
          "color": undefined,
          "cursor": "pointer",
          "letterSpacing": "normal",
        }
      }
      title="Logging in" />
  </div>
</div>
`;

exports[`Login renders correctly when login needs are still unknown 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "color": "rgb(64, 64, 64)",
      "padding": "4px 10px",
    }
  }>
  <i
    className="giu-icon fa fa-circle-o-notch fa-lg fa-fw fa-spin"
    style={
      Object {
        "color": undefined,
        "cursor": "pointer",
        "letterSpacing": "normal",
      }
    } />
</div>
`;

exports[`Login renders correctly when no login is required 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "color": "rgb(64, 64, 64)",
      "padding": "4px 10px",
    }
  }>
  <i>
    No login required to see server logs
  </i>
</div>
`;
